src/codegen/spirv.zig

```diff
@@ -4167,7 +4169,7 @@ const DeclGen = struct {
                         return self.todo("switch on runtime value???", .{});
                     };
                     const int_val = switch (cond_ty.zigTypeTag(mod)) {
-                        .Int => if (cond_ty.isSignedInt(mod)) @as(u64, @bitCast(value.toSignedInt(mod))) else value.toUnsignedInt(mod),
+                        .Bool, .Int => if (cond_ty.isSignedInt(mod)) @as(u64, @bitCast(value.toSignedInt(mod))) else value.toUnsignedInt(mod),
```

Comment:
While the implementation looks good, there is a case I'm not sure about: Though all operations return 1 or 0, the bools are in principle made to work with non-0 and 0. This implementation of switch breaks that "convention" by only switching on 0 and 1. The alternative way to handle that would be to lower switch-on-bool to an if statement. What do you think?